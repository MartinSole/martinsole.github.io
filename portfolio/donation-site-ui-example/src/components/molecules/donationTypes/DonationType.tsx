import { RadioButton } from '../../atoms'

interface Props {
  donationTypes: string[]
  handleDonationType: React.ChangeEventHandler
}

const DonationType = ({ donationTypes, handleDonationType }: Props) => {
  return (
  <fieldset className="gridContainer gridGap">
    <legend className='srOnly'>Choose a donation type</legend>
    {donationTypes.map(((donationType, index) => <RadioButton {...{
      id: `donation-type-${index}`,
      name: 'donation-type',
      text: donationType,
      handleOnChange: handleDonationType,
      index
    }} 
    key={donationType}
    />))}
  </fieldset>)
}

export default DonationType