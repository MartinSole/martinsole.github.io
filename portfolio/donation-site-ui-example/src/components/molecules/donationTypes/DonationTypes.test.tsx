import '@testing-library/jest-dom'
import { fireEvent, render } from "@testing-library/react"
import DonationType from "./DonationType"

test("Renders the DonationAmounts component", () => {
    const mockProps = {
        donationTypes: ['weekly', 'monthly', 'yearly'],
        handleDonationType: jest.fn(),
    }
   const {container} = render(<DonationType donationTypes={mockProps.donationTypes} handleDonationType={mockProps.handleDonationType}/>)
    expect(container).toMatchSnapshot()
    const radioButtons = container.querySelectorAll('input')
    expect(radioButtons).toHaveLength(3)
    fireEvent.click(radioButtons[1])
    expect(mockProps.handleDonationType).toHaveBeenCalled()
})
