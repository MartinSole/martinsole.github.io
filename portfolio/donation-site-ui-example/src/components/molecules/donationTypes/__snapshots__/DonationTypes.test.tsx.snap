// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the DonationAmounts component 1`] = `
<div>
  <fieldset
    class="gridContainer gridGap"
  >
    <legend
      class="srOnly"
    >
      Choose a donation type
    </legend>
    <input
      checked=""
      id="donation-type-0"
      name="donation-type"
      type="radio"
      value="weekly"
    />
    <label
      class="buttonBase"
      for="donation-type-0"
    >
      weekly
    </label>
    <input
      id="donation-type-1"
      name="donation-type"
      type="radio"
      value="monthly"
    />
    <label
      class="buttonBase"
      for="donation-type-1"
    >
      monthly
    </label>
    <input
      id="donation-type-2"
      name="donation-type"
      type="radio"
      value="yearly"
    />
    <label
      class="buttonBase"
      for="donation-type-2"
    >
      yearly
    </label>
  </fieldset>
</div>
`;
